android/src/main/java/com/tailscale/ipn/ui/view/SettingsView.kt

```diff
@@ -44,18 +45,16 @@ import com.tailscale.ipn.ui.util.defaultPaddingModifier
 import com.tailscale.ipn.ui.util.settingsRowModifier
 import com.tailscale.ipn.ui.viewModel.Setting
 import com.tailscale.ipn.ui.viewModel.SettingType
+import com.tailscale.ipn.ui.viewModel.SettingsNav
 import com.tailscale.ipn.ui.viewModel.SettingsViewModel
+import com.tailscale.ipn.ui.viewModel.SettingsViewModelFactory
 
 
-data class SettingsNav(
-    val onNavigateToBugReport: () -> Unit,
-    val onNavigateToAbout: () -> Unit,
-    val onNavigateToMDMSettings: () -> Unit,
-    val onNavigateToManagedBy: () -> Unit,
-)
-
 @Composable
-fun Settings(viewModel: SettingsViewModel) {
+fun Settings(
+    settingsNav: SettingsNav,
+    viewModel: SettingsViewModel = viewModel(factory = SettingsViewModelFactory(settingsNav))
```

Comment:
Notice how we use the `viewModel()` function to manage the lifecycle of the view model, and use `SettingsViewModelFactory` to allow us to inject the `settingsNav`.

Reply:
Wouldn't be Java without a Factory!